src/astro/databases/base.py

```diff
@@ -164,16 +164,52 @@ def populate_table_metadata(self, table: Table) -> Table:
     # ---------------------------------------------------------
     # Table creation & deletion methods
     # ---------------------------------------------------------
-    def create_table(self, table: Table) -> None:
+    def create_table_using_columns(self, table: Table) -> None:
         """
-        Create a SQL table. For this method to work, the table instance must contain columns.
+        Create a SQL table using the table columns.
 
         :param table: The table to be created.
         """
         metadata = table.sqlalchemy_metadata
         sqlalchemy_table = sqlalchemy.Table(table.name, metadata, *table.columns)
         metadata.create_all(self.sqlalchemy_engine, tables=[sqlalchemy_table])
 
+    def create_table_using_schema_autodetection(
+        self, table: Table, file: Optional[File] = None
+    ) -> None:
+        """
+        Create a SQL table, automatically inferring the schema using the given file.
+
+        :param table: The table to be created.
+        :param file: File used to infer the new table columns.
+        """
+        if file is None:
+            raise ValueError(
+                "File is required for creating table using schema autodetection"
+            )
+        source_dataframe = file.export_to_dataframe()  # should we use nrows?
```

Comment:
It doesn't necessarily define the purpose of the load afterwards, @utkarsharma2, since this is being applied only to the first file. But I agree the user can benefit by having a threshold, I'll do this via settings!